src/platforms/esp32/components/avm_sys/platform_nifs.c

```diff
@@ -513,6 +528,47 @@ static term nif_esp_deep_sleep_enable_gpio_wakeup(Context *ctx, int argc, term a
 }
 #endif
 
+static term nif_esp_gpio_wakeup_enable(Context *ctx, int argc, term argv[])
```

Comment:
I would call this `gpio:wakeup_enable/2` to match with the esp-idf function name.

Reply:
this means we are leaking more ESP32 specific stuff into GPIO module. I think they can be made somehow generic, but I'm not sure how.